projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -482,34 +516,40 @@ export class IgxButtonGroupComponent extends DisplayDensityBase implements After
      */
     public _clickHandler(index: number) {
         const button = this.buttons[index];
-        const args: IButtonGroupEventArgs = { cancel: false, owner: this, button, index };
+        const args: IButtonGroupEventArgs = { owner: this, button, index };
+        const cancellableArgs: IButtonGroupCancellableEventArgs = { cancel: false, owner: this, button, index };
 
         if (this.selectionMode !== 'multi') {
             this.buttons.forEach((b, i) => {
                 if (i !== index && this.selectedIndexes.indexOf(i) !== -1) {
-                    this.deselected.emit({ cancel: false, owner: this, button: b, index: i });
+                    this.deselecting.emit({ cancel: false, owner: this, button: b, index: i });
```

Comment:
Yeah, this won't do the trick - this was an issue with the previous attempt for cancellable as well:
This deselection is a side effect of the actual selection of something else, right? Although it's a bit odd in a loop, we're talking single selection and one button active that needs to be deselected because another button is selected.
Anyway, in this case I'd expect you to cancel the primary action (selecting) and this portion only emits the old `deselected` event as a by-product that can't be avoided, otherwise you end up in an invalid state against the more, which I don't think we want to do.
In other words - this entire operation (deselecting an active button) cannot happen before `this.selecting.emit(cancellableArgs);` is emitted and not cancelled. Otherwise, this shouldn't happen at all.

Reply:
If we are not firing deselecting for exclusive selection, you should also add this to the API docs.